modules/core/shared/src/main/scala/scaladex/core/model/Artifact.scala

```diff
@@ -221,29 +221,65 @@ case class Artifact(
     }
 
   // todo: Add tests for this
-  def scastieURL: String = {
-    val tryBaseUrl = "https://scastie.scala-lang.org/try"
+  // def scastieURL: String = {
+  //   val tryBaseUrl = "https://scastie.scala-lang.org/try"
+
+  //   val targetParam = binaryVersion.platform match {
+  //     case ScalaJs(_) => Some("t" -> "JS")
+  //     case _          => None
+  //   }
+
+  //   val scalaVersionParam = binaryVersion.language match {
+  //     case Scala(v) => Some("sv" -> v.toString)
+  //     case _        => None
+  //   }
+
+  //   val params: List[(String, String)] = List(
+  //     "g" -> groupId.value,
+  //     "a" -> artifactName.value,
+  //     "v" -> version.toString
+  //   ) ++ targetParam ++ scalaVersionParam
+
+  //   params
+  //     .map { case (k, v) => s"$k=$v" }
+  //     .mkString(tryBaseUrl + "?", "&", "")
+  // }
+
+  def isSbtPlugin: Boolean = platform match {
+    case SbtPlugin(_) => true
+    case _            => false
+  }
 
-    val targetParam = binaryVersion.platform match {
-      case ScalaJs(_) => Some("t" -> "JS")
-      case _          => None
-    }
+  def getOrganization: String = projectRef.organization.toString
 
-    val scalaVersionParam = binaryVersion.language match {
-      case Scala(v) => Some("sv" -> v.toString)
-      case _        => None
-    }
+  def getRepository: String = projectRef.repository.toString
 
-    val params: List[(String, String)] = List(
-      "g" -> groupId.value,
-      "a" -> artifactName.value,
-      "v" -> version.toString
-    ) ++ targetParam ++ scalaVersionParam
+  def scastieURL: Option[String] =
+    if (isSbtPlugin) {
+      None
+    } else {
+      val tryBaseUrl = "https://scastie.scala-lang.org/try"
 
-    params
-      .map { case (k, v) => s"$k=$v" }
-      .mkString(tryBaseUrl + "?", "&", "")
-  }
+      val targetParam = binaryVersion.platform match {
+        case ScalaJs(_) => Some("t" -> "JS")
+        case _          => Some("t" -> "JVM")
+      }
+
+      val scalaVersionParam = binaryVersion.language match {
+        case Scala(v) => Some("sv" -> v.toString)
+        case _        => None
+      }
+
+      val params: List[(String, String)] = List(
+        "g" -> groupId.value,
+        "a" -> artifactName.value,
+        "v" -> version.toString,
+        "o" -> getOrganization, // get organization value
+        "r" -> getRepository // get repository value
```

Comment:
I am curious about the Scastie endpoint. How did you find out that we need those params?

Reply:
I found it here- https://github.com/scalacenter/scastie/#try-library

In that it said that `o -` scaladex organization and `r -` scaladex repository fields are `required` fields

Reply:
I will replace the getter setters with inline `projectRef.organization.toString` in next commit

I had another doubt, the html code of Scastie box is given and also it is mentioned that we have to add this in `artifact.scala.html` 

But it is not mentioned where actually Scastie box was previously situated inside `artifact.scala.html`

Can you please guide me in this?

Reply:
I don't remember how this Scastie box was.

I would try to add it here to see how it looks. Maybe it is better to nest it in another white box or maybe not.

![image](https://github.com/scalacenter/scaladex/assets/13123162/4912d008-0c20-4a7c-bebe-4eec20d94898)

Reply:
I will add it as you suggested and we can also experiment with it later to find where it fits best

Thank you for your guidance, I will commit with above changes implemented as soon as possible!

Reply:
Hello @adpi2 

This how it looks after adding Scastie box-

![image](https://github.com/scalacenter/scaladex/assets/113872531/6b337a93-4ae5-487b-a0fc-ea5652c939f0)

And this is url you get when clicked on `Try online with scastie box`- https://scastie.scala-lang.org/try?g=com.typesafe.akka&a=akka-actor&v=2.6.19&o=akka&r=akka&t=JVM&sv=3.x

Does this meets your expectations?